core/src/test/scala/unit/kafka/admin/AdminTest.scala

```diff
@@ -110,6 +110,7 @@ class AdminTest extends ZooKeeperTestHarness with Logging with RackAwareTest {
 
   @Test
   def testTopicCreationInZK() {
+    zkClient.makeSurePersistentPathExists(BrokerIdsZNode.path)
```

Comment:
Why is this needed now and it wasn't before?

Reply:
previously zkUtils.registerBrokerInZk -> zkCheckedEphemeral.create was creating the parent path recursively. Now zkClient.registerBrokerInZk expects parent path to present.

testcase for zkClient.registerBrokerInZk:
https://github.com/apache/kafka/blob/trunk/core/src/test/scala/unit/kafka/zk/KafkaZkClientTest.scala#L419

Reply:
`TestUtils.createBrokersInZk` should create the parent path in that case IMO. I agree that `zkClient.registerBrokerInZk` should say as it is.

Reply:
Agree. moved the  parent path creation to  parent path. Updated the PR.